app/src/main/java/com/javahelp/frontend/fragments/ProviderListAdapter.java

```diff
@@ -0,0 +1,82 @@
+package com.javahelp.frontend.fragments;
+
+import android.annotation.SuppressLint;
+import android.app.Activity;
+
+import android.content.Context;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ArrayAdapter;
+import android.widget.BaseAdapter;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import com.javahelp.R;
+import com.javahelp.model.survey.SurveyResponse;
+import com.javahelp.model.user.ProviderUserInfo;
+import com.javahelp.model.user.User;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class ProviderListAdapter extends ArrayAdapter<Map<User, SurveyResponse>> {
+    private final Integer imgID = R.drawable.user48;
+
+    public ProviderListAdapter(Context context, List<Map<User, SurveyResponse>> usersAndResponses) {
+        super(context, R.layout.provider_list, usersAndResponses);
+    }
+
+    @SuppressLint("SetTextI18n")
+    public View getView(int position, View view, ViewGroup parent) {
+        // Check if an existing view is being reused, otherwise inflate the view
+        if (view == null) {
+            view = LayoutInflater.from(getContext()).inflate(R.layout.provider_list, parent, false);
+        }
+
+        TextView titleText = (TextView) view.findViewById(R.id.title);
+        ImageView imageView = (ImageView) view.findViewById(R.id.icon);
+        TextView subtitleCert = (TextView) view.findViewById(R.id.certified);
+        TextView subtitleNumber = (TextView) view.findViewById(R.id.number);
+        TextView subtitleAddr = (TextView) view.findViewById(R.id.address);
+        TextView subtitleAttr0 = (TextView) view.findViewById(R.id.attr0);
+        TextView subtitleAttr1 = (TextView) view.findViewById(R.id.attr1);
+        TextView subtitleAttr2 = (TextView) view.findViewById(R.id.attr2);
+        TextView subtitleAttr3 = (TextView) view.findViewById(R.id.attr3);
+
```

Comment:
Was the databinding not working here ?

Reply:
It should be used. The current version should only be a temporary solutioni

Reply:
Ok I see